### DISCUSSION TITLE
Quadrature counter / rotary encoder wiring

### DISCUSSION THREAD

**Discussion by petersaj** [harp-tech/protocol#99] [Open-ended]
Port2 is designed to act as a quadrature counter - how should we wire an input to use that quadrature counter? 

Specifically, we would like to connect a rotary encoder (Kubler 05.2400.1122.1024). We need to know which pins of the Port2 RJ45 connector should accept the GND/+V/A/B inputs.

**Comment by filcarv** [comment]
The Kubler 05.2400.1122.1024 can use 5 Volts as a supply, so you can connect directly to the Harp Behavior.

The connections would be:
RJ45 (pin 4) - 5 V
RJ45 (pin 5) - GND
RJ45 (pin 2) - DIO - B
RJ45 (pin 1) - DI - A

You can use the Harp Behavior Port Breakout that already has the connections exposed. Just need to connect it to Harp Behavior's Port2 using an ethernet cable.
![image](https://user-images.githubusercontent.com/34546039/212907975-0226b29a-c7f0-46a5-99b7-1ce6bad787c0.png)

In order to enable this feature, you need to write the value 4 to a specific register (ADD_REG_EN_ENCODERS with address 82).
I'm leaving [here](https://github.com/harp-tech/community/files/10436083/QuadratureReader.zip) a Bonsai Workflow that does this for you. Also includes the encoder events and a counter reset that can be asserted by pressing the key Space on your keyboard.

![image](https://user-images.githubusercontent.com/34546039/212929957-f7092804-c9c5-4aca-97b8-23fadd32a40b.png)

**Comment by HeRodrgues** [comment]
Hi guys, quick question: could I use the same bonsai workflow and a similar connection to the Poke Breakout with this flowmeter?
https://www.ato.com/Content/doc/ATO-MF4000-gas-mass-flow-meter-user-manual.pdf
